utils/linux.go

```diff
@@ -203,3 +232,50 @@ func GetLinuxHeaderPath(distro string) (string, error) {
 
 	return path, nil
 }
+
+func KernelSupportsIDMappedMounts() (bool, error) {
+	var major, minor int
+
+	rel, err := GetKernelRelease()
+	if err != nil {
+		return false, err
+	}
+
+	major, minor, err = ParseKernelRelease(rel)
+	if err != nil {
+		return false, err
+	}
+
+	// ID-Mapped mounts requires Linux kernel >= 5.12
+
+	if major < 5 {
+		return false, nil
+	} else if major == 5 && minor < 12 {
+		return false, nil
+	} else {
+		return true, nil
+	}
+}
+
+// KernelModSupported returns nil if the given module is loaded in the kernel.
+func KernelModSupported(mod string) (bool, error) {
+
+	// Load the module
+	exec.Command("modprobe", mod).Run()
+
+	// Check if the module is in the kernel
```

Comment:
Why isn't it enough to check `modprobe` return value to save us the cycles of reading `/proc/modules` (usually a 100+ lines file)?

Reply:
Good point. At the time, I wanted to remove the dependency on the `modprobe` command and get the info directly from the kernel (which explains why we don't check the output of modprobe in the code). I guess we could rely solely on modprobe, but it's not clear to me if its included in all distros (though I suspect it is).